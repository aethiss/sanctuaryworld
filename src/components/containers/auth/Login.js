import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Next
import Link from 'next/link'
import Image from 'next/image'

// Formik
import {Formik, Form} from 'formik'
import * as Yup from 'yup'

// Components
import BodyContainer from '../../commons/wrapper/BodyContainerResponsive'
import FormikFields from '../../commons/forms/FormikFields'
import DiabloResponsiveButton from '../../commons/buttons/DiabloResponsiveButton/DiabloResponsiveButton'
import { getUser } from '../../../redux/handlers/authActions'


const styles = {
  contentContainer: {
    textAlign: 'center',
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  button: {
    marginTop: '35px',
  },
  separator: {
    marginTop: '10px',
  },
  overflow: {
    overflow: 'hidden',
  },
}

@connect(() => ({}), {
  actionLogin: getUser,
})
class Login extends Component  {

  render() {
    const backgroundColor = '#131210'
    const { actionLogin } = this.props


    return (
      <BodyContainer
        background='/images/background/act1-bg.jpg'
        contentOnCenter
      >
        <div style={styles.contentContainer}>
          <span>
            <Image
              src="/images/background/d4_logo.png"
              alt="diablo 4 logo"
              width={300}
              height={140}
              style={styles.overflow}
            />
          </span>
          <>
            <Formik
              initialValues={{email: '', password: ''}}
              validationSchema={Yup.object({
                email: Yup.string()
                  .email('Invalid email address')
                  .required('Required'),
                password: Yup.string()
                  .max(25, 'Must be 25 characters or less')
                  .min(8, 'Must be 8 characters or more')
                  .required('Required'),
              })}
              onSubmit={(values, {setSubmitting}) => {
                if (typeof actionLogin === 'function') {
                  actionLogin(values)
                } else {
                  setSubmitting(false)
                }
              }}>
              {props => (
                <Form style={{...styles.form}}>
                  <FormikFields
                    name="email"
                    type="text"
                    label={`Sanctuary Account Email`}
                    formikProps={props}
                    icon="fa-at"
                    backgroundColor={backgroundColor}
                  />
                  <FormikFields
                    name="password"
                    type="password"
                    label="Password"
                    formikProps={props}
                    icon="fa-key"
                    backgroundColor={backgroundColor}
                  />
                  <div style={styles.button}>
                    <DiabloResponsiveButton text="Sign In" minWidth={170} />
                  </div>
                </Form>
              )}
            </Formik>
          </>
          <div style={styles.separator}>
            or <br/>
            <Link href='/signin'>
              <a>register</a>
            </Link>
          </div>
        </div>
      </BodyContainer>
    )
  }
}

Login.propTypes = {
  battletag: PropTypes.string,
  actionLogin: PropTypes.func,
}

Login.defaultProps = {
  battletag: '',
  actionLogin: () => false,
}

export default Login
