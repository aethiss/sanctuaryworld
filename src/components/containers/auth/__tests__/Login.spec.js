import React from 'react'
import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

import Login from '../Login'

jest.mock('next/image', () => (jest.fn(() => null)))

const mockProps = {
  user: {
    battletag: 'pippo#1234',
    action: jest.fn(),
  }
}

describe('Login', () => {
  describe('Render', () => {
    it('[Login] : Render default', () => {
      const store = mockStore(mockProps)
      const testRenderer = TestRenderer.create(
        <Provider store={store}>
          <Login />
        </Provider>
      )
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
