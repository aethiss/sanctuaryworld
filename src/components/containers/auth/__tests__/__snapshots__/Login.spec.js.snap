// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Render [Login] : Render default 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
  style={
    Object {
      "backgroundImage": "/images/background/act1-bg.jpg",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span />
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginTop": "30px",
            "width": "95%",
          }
        }
      >
        <label
          htmlFor="email"
          style={
            Object {
              "fontSize": "18px",
              "paddingBottom": "7px",
            }
          }
        >
          Sanctuary Account Email
        </label>
        <div
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <i
            className="fa fa-at"
            style={
              Object {
                "justifySelf": "center",
              }
            }
          />
          <input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            style={
              Object {
                "backgroundColor": "#131210",
                "border": "2px solid #ccc",
                "boxShadow": "0px 0px 9px 3px #ccc",
                "color": "#FFF",
                "height": "35px",
                "opacity": "0.7",
                "width": "275px",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginTop": "30px",
            "width": "95%",
          }
        }
      >
        <label
          htmlFor="password"
          style={
            Object {
              "fontSize": "18px",
              "paddingBottom": "7px",
            }
          }
        >
          Password
        </label>
        <div
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <i
            className="fa fa-key"
            style={
              Object {
                "justifySelf": "center",
              }
            }
          />
          <input
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            style={
              Object {
                "backgroundColor": "#131210",
                "border": "2px solid #ccc",
                "boxShadow": "0px 0px 9px 3px #ccc",
                "color": "#FFF",
                "height": "35px",
                "opacity": "0.7",
                "width": "275px",
              }
            }
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "marginTop": "35px",
          }
        }
      >
        <button
          className="makeStyles-button-2"
        >
          <span
            className="makeStyles-buttonLeft-4"
          />
          <span
            className="makeStyles-buttonMid-3"
          >
            <span
              className="makeStyles-buttonText-6"
            >
              Sign In
            </span>
          </span>
          <span
            className="makeStyles-buttonRight-5"
          />
        </button>
      </div>
    </form>
    <div
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      or 
      <br />
      <a
        href="/signin"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        register
      </a>
    </div>
  </div>
</div>
`;
